### PR TITLE
complete java exercises day 3 and day 4

### PR DESCRIPTION
**Pull request by joshlee2ey**
complete exercises 8/9/10

t.com/113386053/192046507-6eba5071-be0f-4d0e-919e-8edffaa579b0.PNG)
mixlist

### CODE REVIEW

File: joshlee/java/exercise9.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import java.io.File;  // Import the File class
+import java.io.FileNotFoundException;  // Import this class to handle errors
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner; // Import the Scanner class to read text files
+import java.util.Collections;
+
+public class exercise9 {
+    public static void main(String[] args){
+        File obj = new File("C:\\Users\\VN992XV\\Documents\\git\\cdp\\joshlee\\java\\words.txt");
```
Comment (madhanramu) [inline joshlee/java/exercise9.java:16]: have this as a Class constant "C:\\Users\\VN992XV\\Documents\\git\\cdp\\joshlee\\java\\words.txt")

File: joshlee/java/ShoppingCart.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            throw new Exception("Product was not found in cart and cannot be removed.");
+        }
+    }
+
+    /**
+     * numberOfItemsInCart method counts total items in cart
+     * @param products product list to count items from 
+     * @return total count of items in cart
+     */
+    public int numberOfItemsInCart(List<Product> products){
```
Comment (madhanramu) [inline joshlee/java/ShoppingCart.java:69]: change name getNumberOfItemsInCart()

File: joshlee/java/ShoppingCart.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        products.add(p);
+    }
+
+    /**
+     * getCartBalance method which adds products' prices
+     * @param products list of products which will add to total balance
+     * @return sum of products' prices
+     */
+    public int getCartBalance(List<Product> products){
+        int balance = 0;
```
Comment (madhanramu) [inline joshlee/java/ShoppingCart.java:35]: No need to pass param `List<Product> products`\

use this.products inside the method

File: joshlee/java/ShoppingCart.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    public ShoppingCart(List<Product> products){
+        this.products = products;
+    } 
+
+    /**
+     * addToCart method which adds products to list
+     * @param products product list to add into
+     * @param p product to be added
+     */
+    public void addToCart(List<Product> products, Product p){
```
Comment (madhanramu) [inline joshlee/java/ShoppingCart.java:25]: no need to pass `List<Product>` products

File: joshlee/java/ShoppingCart.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        }
+        return balance;
+    }
+
+    /**
+     * removeItem method which removes product p from product list
+     * @param products list to remove product from 
+     * @param p item to remove
+     */
+    public void removeItem(List<Product> products, String productTitle) throws Exception{
```
Comment (madhanramu) [inline joshlee/java/ShoppingCart.java:47]: no need to pass `List<Product> products.` 
use this.products inside the method 
follow this in all the methods

**Review by madhanramu — CHANGES_REQUESTED** [review]

File: joshlee/java/exercise9.java (hunk 1 above)
Comment (joshlee2ey) [reply joshlee/java/exercise9.java:16]: do you mean keep it outside the main method? as a global constant?

**Review by madhanramu — APPROVED** [review]

**Pull request merged**